File: Database/games.js
const dbhandler = require("./adapters/dbhandler.js");
const apicomm = require("./adapters/apihandler.js");

async function run(startDate = "2021-10-01", endDate = "2022-05-01") {
  const db = new dbhandler.Database();
  await db.init();

  const games = await apicomm.nhlApiRequest(
    `/api/v1/schedule?startDate=${startDate}&endDate=${endDate}`
  );
  const { dates } = games;
  let updatedGames = 0;
  console.log("Skipping games that are already finished...");
  try {
    const collection = db.getCollection("games");
    for (let date of dates) {
      for (let game of date.games) {
        const options = { upsert: true };
        const filter = { gamePk: game.gamePk };
        const updateDoc = {
          $set: {
            gameType: game.gameType,
            season: game.season,
            date: date.date,
            gameDate: new Date(game.gameDate),
            home: game.teams.home,
            away: game.teams.away,
            status: game.status,
            venue: game.venue,
            lastUpdate: new Date(),
          },
        };
        try {
          const queryResult = await collection.updateOne(
            filter,
            updateDoc,
            options
          );
          console.log(
            `${queryResult.matchedCount} document(s) matched the filter, updated ${queryResult.modifiedCount} document(s): ${game.gamePk}`
          );
          updatedGames++;
        } catch (ex) {
          console.log(ex);
        }
      }
    }
  } finally {
    await db.closeClient();
    console.log(`Updated ${updatedGames} game(s).`);
  }
}

exports.handler = async (event) => {
  await run(event.startDate, event.endDate);
};

run().catch(console.dir);

File: Database/games.py
from datetime import datetime
from apihandler import nhl_api_request
from dbhandler import get_collection
import sys

start_date = "2022-10-01"
end_date = "2023-05-01"
if len(sys.argv) > 2:
  start_date = sys.argv[1]
  end_date = sys.argv[2]

collection = get_collection("games")

all_games = nhl_api_request("/api/v1/schedule", {"startDate": start_date, "endDate": end_date})
dates = all_games["dates"]

updated_games = 0
for date in dates:
  games = date["games"]
  for game in games:
    filter = { "gamePk": game["gamePk"] }
    update = {
          "$set": {
            "gameType": game["gameType"],
            "season": game["season"],
            "date": date["date"],
            "gameDate": datetime.strptime(game["gameDate"], '%Y-%m-%dT%H:%M:%SZ'),
            "home": game["teams"]["home"],
            "away": game["teams"]["away"],
            "status": game["status"],
            "venue": game["venue"],
            "lastUpdate": datetime.now()
          }
        }
    result = collection.update_one(filter, update, True)
    print("{0} games found, {1} inserted or updated. Game Id: {2} ".format(result.matched_count, result.modified_count, game["gamePk"]))
    updated_games+=1

print("updated games: {0}".format(updated_games))
